tests/index.test.ts

```diff
@@ -97,14 +112,55 @@ const testSchema = {
             "format": "date-time",
             "description": "A datetime object, we will automatically serialize and deserialize\nthis for you.",
             "nullable": true
+          },
+          "aMapOfMap": {
+            "type": "object",
+            "additionalProperties": {
+              "type": "object",
+              "additionalProperties": {
+                "$ref": "#/components/schemas/EmbeddedObject"
+              }
+            }
           }
         }
       },
       "MapSchema": {
+        "description": "A map schema",
+        "type": "object",
+        "additionalProperties": {
+          "type": "string"
+        }
+      },
+      "MapOfMapSchema": {
+        "description": "A map of map schema",
+        "type": "object",
+        "additionalProperties": {
+          "type": "object",
           "additionalProperties": {
-              "type": "string"
+            "type": "string"
           }
+        }
+      },
```

Comment:
I'm fine with this, but do we really need to support recursive types? I'm okay with denying this pattern if it complicates the bindgens.

Reply:
I think it would be pretty rare. If you have deeply nested, types maps, that's probably best being an object.

Reply:
I think I am going to do that, this should cover most of the use cases while keeping the bindgens sane